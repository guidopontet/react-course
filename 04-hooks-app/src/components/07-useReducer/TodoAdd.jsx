import React from 'react'

import useForm from '../../hooks/useForm'

export const TodoAdd = ({ handleAddTodo }) => {
  const [ { description }, handleInputChange, resetForm ] = useForm({
    description: '',
  })

  const handleSubmit = (e) => {
    e.preventDefault();

    if (description.trim().length < 1 ) { return; }

    const newTodo = {
      id: new Date().getTime(),
      description: description,
      done: false,
    };

    const action = {
      type: 'ADD',
      payload: newTodo,
    }

    handleAddTodo(action);
    resetForm();
  }

  return (
    <>
      <h4>Agregar TODO</h4>
      <hr />

      <form onSubmit={ handleSubmit }>
        <input type="text"
                className='form-control'
                name='description'
                placeholder='Ingresa la tarea'
                autoComplete='off'
                value={ description }
                onChange={ handleInputChange }/>

        <button type='submit'
                className='btn btn-primary btn-block mt-3'>
          Agregar
        </button>
      </form>
    </>
  )
}
