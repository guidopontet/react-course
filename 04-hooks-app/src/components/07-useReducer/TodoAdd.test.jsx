import { shallow } from "enzyme";

import { TodoAdd } from "./TodoAdd";

describe('TodoAdd', () => {
  const handleAddTodo = jest.fn();

  const wrapper = shallow(
    <TodoAdd
      handleAddTodo={ handleAddTodo }
    />
  );

  test('should render', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('should not call handleAddTodo', () => {
    const formSubmit = wrapper.find('form').prop('onSubmit');

    formSubmit({ preventDefault() {} });

    expect(handleAddTodo).not.toHaveBeenCalled();
  })

  test('should call handleAddTodo', () => {
    const value = 'Tarea 1';
    wrapper.find('input').simulate('change', { target: { value, name: 'description' } });

    const formSubmit = wrapper.find('form').prop('onSubmit');

    formSubmit({ preventDefault() {} });

    expect(handleAddTodo).toHaveBeenCalled();

    // Reset
    expect(wrapper.find('input').prop('value')).toBe('');
  })
})
