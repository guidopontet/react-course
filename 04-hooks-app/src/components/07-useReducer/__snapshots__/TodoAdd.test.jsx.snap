// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoAdd should render 1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Ingresa la tarea"
      type="text"
      value=""
    />
    <button
      className="btn btn-primary btn-block mt-3"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
